ui/src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -274,7 +274,7 @@ export const ApplicationsList = (props: RouteComponentProps<{}>) => {
                                                         <Query>
                                                             {q => (
                                                                 <div className='applications-list__search'>
-                                                                    <i className='fa fa-search' />
+                                                                    <i className='fa fa-search' onClick={() => document.getElementById('search-bar').focus()} />
```

Comment:
Please use [React.useRef](https://reactjs.org/docs/hooks-reference.html#useref) instead of `document.getElementById('search-bar')`

Reply:
Thanks for the review. With `useRef` I'm able to access the element. But Autocomplete component is throwing an error. Should check what's causing this

```
Autocomplete.js:335 Uncaught TypeError: Cannot read property 'ownerDocument' of undefined
    at Autocomplete.isInputFocused (Autocomplete.js:335)
    at Autocomplete.handleInputClick (Autocomplete.js:341)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:270)
    at executeDispatch (react-dom.development.js:561)
    at executeDispatchesInOrder (react-dom.development.js:583)
    at executeDispatchesAndRelease (react-dom.development.js:680)
    at executeDispatchesAndReleaseTopLevel (react-dom.development.js:688)
```

Reply:
Thanks for catching it @chetan-rns ! I think this is happening because `Autocomplete` itself uses `ref` property of input and it makes it difficult to get input reference.

The best workaround I found is: get the reference to search input parent and use `querySelector` to find input:

```html
const searchBar = React.useRef<HTMLDivElement>(null);
...
<div className='applications-list__search' ref={searchBar}>
    <i
        className='fa fa-search'
        onClick={() => {
            if (searchBar.current) {
                searchBar.current.querySelector('input').focus();
            }
        }}
    />
...
```

This is far from perfect but at least we don't use hard coded id.

Reply:
I couldn't think of anything better. This approach is better than using `id`. Thank you.